File: go/internal/scoring/testdata/structural-v1/typescript/state_coherent_before.ts
export class CoherentBefore {
  private total = 0;

  advance(delta: number): number {
    if (delta < 0) return this.total;
    this.total += delta;
    if (this.total > 100) this.total = 100;
    return this.total;
  }

  advanceAgain(delta: number): number {
    if (delta < 0) return this.total;
    this.total += delta;
    if (this.total > 100) this.total = 100;
    return this.total;
  }
}

File: go/internal/scoring/testdata/structural-v1/rust/state_coherent_before.rs
pub struct CoherentBefore { total: i32 }

impl CoherentBefore {
    pub fn advance(&mut self, delta: i32) -> i32 {
        if delta < 0 { return self.total; }
        self.total += delta;
        if self.total > 100 { self.total = 100; }
        self.total
    }

    pub fn advance_again(&mut self, delta: i32) -> i32 {
        if delta < 0 { return self.total; }
        self.total += delta;
        if self.total > 100 { self.total = 100; }
        self.total
    }
}
